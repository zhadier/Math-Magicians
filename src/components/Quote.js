import { useEffect, useState } from 'react';
import './Quote.scss';

const Quote = () => {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState('');
  useEffect(() => {
    setLoading(true);
    const getQuote = async () => {
      const connect = await fetch('https://random-math-quote-api.herokuapp.com/');
      const newQuote = await connect.json();
      setQuote(`${newQuote.quote} ~${newQuote.author}`);
      setLoading(false);
    };
    getQuote();
  }, [setQuote]);
  return (
    <section className="quote-section">
      <p>{loading ? 'Loading...' : quote }</p>
    </section>
  );
};

export default Quote;
