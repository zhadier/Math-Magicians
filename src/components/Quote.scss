.quote-section {
  align-items: center;
  justify-content: center;
  padding: 50px;

  p {
    padding: 20px;
    font-size: 1.7rem;
    font-style: italic;
    border: 1px solid;
    border-radius: 10px;
    text-align: center;
    box-shadow: 0 0 10px blue;
    animation: 5s infinite linear quote;
  }

  @keyframes quote {
    0% {
      filter: hue-rotate(0deg);
    }

    100% {
      filter: hue-rotate(360deg);
    }
  }
}
