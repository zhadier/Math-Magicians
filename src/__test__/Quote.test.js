import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

it('quotes render correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();

  expect(tree).toMatchSnapshot();
});
