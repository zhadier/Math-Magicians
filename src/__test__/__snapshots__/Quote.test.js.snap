// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quotes render correctly 1`] = `
<section
  className="quote-section"
>
  <p>
    
  </p>
</section>
`;
